### PR TITLE
Add secondary pages

### PR DESCRIPTION
**Pull request by BaptisteLegat**

### CODE REVIEW

File: src/pages/contact/ContactIconsList.tsx
Comment (Dercraker) [inline src/pages/contact/ContactIconsList.tsx:1]: Move it in components/folderNameOfComponent/ComponentName.tsx

File: src/pages/contact/ContactIcons.scss
Comment (Dercraker) [inline src/pages/contact/ContactIcons.scss:1]: move it in component folder

File: src/pages/contact/ContactPage.tsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+      </Title>
+      <Paper shadow="md" radius="lg" mt="xl" p="xl">
+        <div className="contact">
+          <div className="contact__contacts">
+            <Text className="contact__title" fz="lg" fw={700} c="#fff">
+              Information de contact
+            </Text>
+            <ContactIconsList />
+          </div>
+          <form
```
Comment (Dercraker) [inline src/pages/contact/ContactPage.tsx]: Move the form in component

File: src/pages/contact/ContactPage.tsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+      <Title order={1} mb="lg" ta="center" mt="lg">
+        Contactez-nous
+      </Title>
+      <Paper shadow="md" radius="lg" mt="xl" p="xl">
+        <div className="contact">
+          <div className="contact__contacts">
+            <Text className="contact__title" fz="lg" fw={700} c="#fff">
+              Information de contact
+            </Text>
+            <ContactIconsList />
```
Comment (Dercraker) [inline src/pages/contact/ContactPage.tsx:18]: the height do not follow the parent

File: src/pages/notice/NoticePage.tsx [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+    date: '2022-11-15',
+    comment:
+      'Je suis très satisfaite de ma réservation, je n’hésiterai pas à revenir !',
+    rating: 5,
+  },
+];
+
+const NoticePage = () => {
+  const [comment, setComment] = useState('');
+  const [rating, setRating] = useState(0);
```
Comment (Dercraker) [inline src/pages/notice/NoticePage.tsx]: add type for State like this : 
```suggestion
  const [comment, setComment] = useState<string>('');
  const [rating, setRating] = useState<number>(0);
```

File: src/pages/notice/NoticePage.tsx [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+                  value={review.rating}
+                  readOnly
+                  title="Note attribuée par le client"
+                />
+              </Group>
+            </Card>
+          </Carousel.Slide>
+        ))}
+      </Carousel>
+      <form onSubmit={handleSubmit}>
```
Comment (Dercraker) [inline src/pages/notice/NoticePage.tsx]: in component

**Review by Dercraker — CHANGES_REQUESTED** [review]

File: src/components/cardCustom/featureCard/FeatureCard.tsx [hunk 5]
```diff
@@ -1,18 +1,14 @@
...
   title: string;
   description: string;
 }
 
-export function DestinationCard({
-  icon,
-  title,
-  description,
-}: DestinationCardProps) {
+export function FeatureCard({ icon, title, description }: FeatureCardProps) {
```
Comment (Dercraker) [inline src/components/cardCustom/featureCard/FeatureCard.tsx]: [No function, only arrow func](https://www.w3schools.com/js/js_arrow_function.asp)

```suggestion
export const FeatureCard = ({ icon, title, description }: FeatureCardProps)  => {
```

Comment (Dercraker) [inline src/components/cardCustom/featureCard/FeatureCard.tsx]: The variables are taken from your component's props objects : 
```suggestion
export const FeatureCard = (props: FeatureCardProps) => {
```
And call props.icon, props.title, props.description

File: src/components/contact/ContactIconsList.tsx [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+  title: React.ReactNode;
+  description: React.ReactNode;
+}
+
+function ContactIcon({
+  icon: Icon,
+  title,
+  description,
+  ...others
+}: ContactIconProps) {
```
Comment (Dercraker) [inline src/components/contact/ContactIconsList.tsx]: [No function, only arrow func](https://www.w3schools.com/js/js_arrow_function.asp)
And use props defenition, not props object in func argument

```suggestion
const ContactIcon = (props : ContactIconProps) => {
```

File: src/components/contact/ContactIconsList.tsx [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+}
+
+const data = [
+  { title: 'Email', description: 'contact@fromA2B.com', icon: IconAt },
+  { title: 'Téléphone', description: '+33 1 23 45 67 89', icon: IconPhone },
+  { title: 'Adresse', description: 'Paris, France', icon: IconMapPin },
+  { title: 'Horaire', description: 'Lun - Ven: 9:00 - 18:00', icon: IconSun },
+];
+
+export function ContactIconsList() {
```
Comment (Dercraker) [inline src/components/contact/ContactIconsList.tsx]: [No function, only arrow func](https://www.w3schools.com/js/js_arrow_function.asp)
```suggestion
export const ContactIconsList = () => {
```

File: src/components/contact/FormContact.tsx [hunk 8]
```diff
@@ -0,0 +1,39 @@
+import { Button, Group, SimpleGrid, Text, TextInput, Textarea } from "@mantine/core";
+
+const FormContact = () => {
+    return (
+        <form
+        className="contact__form"
+        onSubmit={(event) => event.preventDefault()}>
```
Comment (Dercraker) [inline src/components/contact/FormContact.tsx]: Add func on this event : 
```suggestion
        onSubmit={handleSubmitForm}>
```

File: src/components/contact/FormContact.tsx [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+            <Button className="contact__control" type="submit">
+              Envoyer
+            </Button>
+          </Group>
+        </div>
+      </form>
+    )
+};
+
+export default FormContact;
```
Comment (Dercraker) [inline src/components/contact/FormContact.tsx]: - export for component
- export default for page and layout
```suggestion
export FormContact;
```

File: src/components/feedback/FormFeedback.tsx
Comment (Dercraker) [inline src/components/feedback/FormFeedback.tsx:1]: [do use mantine UseForm](https://mantine.dev/form/use-form/)

File: src/router/Router.tsx [hunk 10]
```diff
@@ -29,6 +33,11 @@ const Router = () => {
           }
         />
         <Route path="/*" element={<Navigate to="/" />} />
+        <Route path="/feedback" element={<FeedbackPage />} />
+        <Route path="/profile" element={<Navigate to="/profile" />} />
```
Comment (Dercraker) [inline src/router/Router.tsx]: ???? why navigate and not PageElement ?

File: src/components/contact/ContactIconsList.tsx [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+      </Box>
+      <div>
+        <Text size="xs" className="title">
+          {props.title}
+        </Text>
+        <Text className="description">{props.description}</Text>
+      </div>
+    </div>
+  );
+}
```
Comment (Dercraker) [inline src/components/contact/ContactIconsList.tsx]: This ContactIcon will never can be used at another location ? if yes move it in component

File: src/components/contact/ContactIconsList.tsx [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+
+const data = [
```
Comment (Dercraker) [inline src/components/contact/ContactIconsList.tsx]: give a human comprehensible name for this variable

File: src/components/contact/FormContact.tsx [hunk 13]
```diff
@@ -0,0 +1,90 @@
...
+import { zodResolver } from 'mantine-form-zod-resolver';
+import { z } from 'zod';
+
+export const FormContact = () => {
+  const contactSchema = z.object({
+    email: z.string().email("l'email n'est pas valide"),
+    message: z
+      .string()
+      .min(10, 'le message doit contenir au moins 10 caractères'),
+  });
```
Comment (Dercraker) [inline src/components/contact/FormContact.tsx]: No validation on the name or subject of mail ? 

Bug -> it's possible to send form, without name and subject : 
image

File: src/components/contact/FormContact.tsx [hunk 14]
```diff
@@ -0,0 +1,90 @@
...
+      name: '',
+      email: '',
+      subject: '',
+      message: '',
+    },
+
+    validate: zodResolver(contactSchema),
+  });
+
+  const isContactButtonDisabled = !contactForm.isValid();
```
Comment (Dercraker) [inline src/components/contact/FormContact.tsx]: this variable is useless delete it

File: src/components/contact/FormContact.tsx [hunk 15]
```diff
@@ -0,0 +1,90 @@
...
+          minRows={3}
+          resize="both"
+          required
+          {...contactForm.getInputProps('message')}
+        />
+        <Group justify="flex-end" mt="md">
+          <Button
+            className="contact__control"
+            type="submit"
+            disabled={isContactButtonDisabled}>
```
Comment (Dercraker) [inline src/components/contact/FormContact.tsx]:
```suggestion
            disabled={!contactForm.isValid()}>
```

File: src/components/contact/FormContact.tsx [hunk 16]
```diff
@@ -0,0 +1,90 @@
...
+
+  const handleSubmit = async (event: React.FormEvent<HTMLButtonElement>) => {
+    event.preventDefault();
+    throw new Error('Not implemented');
+  };
+
+  return (
+    <form
+      className="contact__form"
+      onSubmit={handleSubmit}
```
Comment (Dercraker) [inline src/components/contact/FormContact.tsx:60]: You have an error of type on this line

image

File: src/components/feedback/FormFeedback.tsx [hunk 17]
```diff
@@ -0,0 +1,68 @@
...
+    validateInputOnChange: true,
+    initialValues: {
+      comment: '',
+      rating: 0,
+    },
+
+    validate: zodResolver(feedbackSchema),
+  });
+
+  const isButtonDisabled = !feedbackForm.isValid();
```
Comment (Dercraker) [inline src/components/feedback/FormFeedback.tsx]: Same as precedent

File: src/components/feedback/FormFeedback.tsx [hunk 18]
```diff
@@ -0,0 +1,68 @@
...
+        placeholder='Ex: "J’ai adoré mon séjour, je recommande vivement !"'
+        {...feedbackForm.getInputProps('comment')}
+      />
+      <Center mt="sm">
+        <StarLikeComponent ChangeRating={handleChangeRating} />
+      </Center>
+      <Center mt="xl">
+        <Button
+          type="submit"
+          disabled={isButtonDisabled}
```
Comment (Dercraker) [inline src/components/feedback/FormFeedback.tsx]: Same as precedent

File: src/router/Router.tsx [hunk 19]
```diff
@@ -22,6 +26,10 @@ const Router = () => {
         <Route path="/" element={<LandingPage />} />
         <Route path="/test" element={<TestPage />} />
         <Route path="/*" element={<Navigate to="/" />} />
+        <Route path="/feedback" element={<FeedbackPage />} />
+        <Route path="/cgu" element={<CguPage />} />
+        <Route path="/contact" element={<ContactPage />} />
+        <Route path="/faq" element={<FaqPage />} />
```
Comment (Dercraker) [inline src/router/Router.tsx:30]: move it in front of line 28 `<Route path="/*" element={<Navigate to="/" />} />`

**Review by Dercraker — CHANGES_REQUESTED** [review]

File: src/pages/feedback/FeedbackPage.tsx [hunk 20]
```diff
@@ -0,0 +1,115 @@
...
+      </Text>
+      <Center mt="xl">
+        <Rating
+          value={4.5}
+          readOnly
+          fractions={2}
+          color="teal"
+          size={60}
+          title="Note moyenne"
+        />
```
Comment (Dercraker) [inline src/pages/feedback/FeedbackPage.tsx]: This value isn't valid and is not the value of comment.

**Review by Dercraker — CHANGES_REQUESTED** [review]

File: src/components/contact/ContactIcon.tsx [hunk 21]
```diff
@@ -0,0 +1,26 @@
+import { Text, Box } from '@mantine/core';
+import { IconSun} from '@tabler/icons-react';
+import './ContactIcons.scss';
+
+interface ContactIconProps
```
Comment (Dercraker) [inline src/components/contact/ContactIcon.tsx:5]: why the props aren't exported. they never used ?

File: src/components/contact/FormContact.tsx [hunk 22]
```diff
@@ -0,0 +1,90 @@
...
+        />
+        <Textarea
+          mt="md"
+          label="Message"
+          placeholder="Merci de nous laisser un message"
+          minRows={3}
+          resize="both"
+          required
+          {...contactForm.getInputProps('message')}
+        />
```
Comment (Dercraker) [inline src/components/contact/FormContact.tsx]: Didn't we talk about a character counter on this textArea?

File: src/components/feedback/FormFeedback.tsx [hunk 23]
```diff
@@ -0,0 +1,67 @@
...
+          variant="filled"
+          mb="lg">
+          Envoyer
+        </Button>
+      </Center>
+    </form>
+  );
+};
+
+export default FormFeedback;
```
Comment (Dercraker) [inline src/components/feedback/FormFeedback.tsx]: Export for component 
Export Default for Pages and Layouts

**Comment by Dercraker** [comment]
Fix the conflicts before complete PR

**Review by Dercraker — CHANGES_REQUESTED** [review]

File: src/components/contact/ContactIcon.tsx (hunk 21 above)
Comment (BaptisteLegat) [reply src/components/contact/ContactIcon.tsx:5]: Never used so don't need to export (and not useful)

**Review by Dercraker — APPROVED** [review]

File: src/components/contact/FormContact.tsx [hunk 24]
```diff
@@ -0,0 +1,116 @@
...
+        <Textarea
+          mt="md"
+          label="Message"
+          placeholder="Merci de nous laisser un message"
+          minRows={3}
+          resize="both"
+          required
+          {...contactForm.getInputProps('message')}
+          onChange={handleMessageChange}
+        />
```
Comment (Dercraker) [inline src/components/contact/FormContact.tsx]: Resize vertical only

File: src/components/contact/FormContact.tsx [hunk 25]
```diff
@@ -0,0 +1,116 @@
...
+          placeholder="Sujet de votre message"
+          required
+          {...contactForm.getInputProps('subject')}
+        />
+        <Textarea
+          mt="md"
+          label="Message"
+          placeholder="Merci de nous laisser un message"
+          minRows={3}
+          resize="both"
```
Comment (Dercraker) [inline src/components/contact/FormContact.tsx]: if user is on mobile he can't use resize

File: src/components/header/LoginHeader.tsx [hunk 26]
```diff
@@ -12,7 +12,7 @@ const LoginHeader = () => {
     <>
       <Group h="100%" mx="50%">
         <img
-          src="/public/vite.svg"
```
Comment (Dercraker) [inline src/components/header/LoginHeader.tsx:15]: Move the svg in public

File: src/components/header/DefaultHeader.tsx [hunk 27]
```diff
@@ -59,7 +59,7 @@ const DefaultHeader = (props: DefaultHeaderProps) => {
           size="sm"
         />
         <Image
-          src="/public/vite.svg"
```
Comment (Dercraker) [inline src/components/header/DefaultHeader.tsx:62]: Move the svg in public

File: src/components/header/DefaultHeader.tsx [hunk 28]
```diff
@@ -59,7 +59,7 @@ const DefaultHeader = (props: DefaultHeaderProps) => {
...
+          src="/vite.svg"
           alt="Logo du site"
           onClick={handleClickLogo}
           className="cursor-pointer"
```
Comment (Dercraker) [inline src/components/header/DefaultHeader.tsx]: Change this by html

File: src/pages/cgu/CguPage.tsx
Comment (Dercraker) [inline src/pages/cgu/CguPage.tsx:1]: edit this page

File: src/pages/landing/Content.tsx
Comment (Dercraker) [inline src/pages/landing/Content.tsx:1]: Change the filename by ContentPage if this is a page or move it in components

**Review by Dercraker — CHANGES_REQUESTED** [review]

**Pull request merged**